---
uint32[] resolutions
int32 return_code